### DISCUSSION TITLE
When run docker command to start the open-webui without internet, one issue occurred.

### DISCUSSION THREAD

**Discussion by quzhixue-Kimi** [open-webui/open-webui#2335] [General]
Hi there,

I just download, export and import the docker image from one ubuntu 20.04 to a new ubuntu 20.04 server. The new ubuntu 20.04 server has no internet.

Based on the doc with below command:
docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main

The containers ran, but wait for a moment. It restarted. I ran docker logs -f open-webui to check the log:
Here is the screen shot of the log. It seems that the container would like to download and load some models from huggingface.co based on internet.

<img width="964" alt="Screenshot 2024-05-17 at 16 53 28" src="https://github.com/open-webui/open-webui/assets/8235746/a09049dd-5070-40ca-8be1-9c52c3d1649c">

It is appreciated that you may tell me whether there is any environment variable to be set without internet or not?

BR
Kimi

**Comment by justinh-rahb** [comment]
```ini
WHISPER_MODEL_AUTO_UPDATE=False
RAG_EMBEDDING_MODEL_AUTO_UPDATE=False
```

**Reply by quzhixue-Kimi** [reply]
Hi there,

Thank you for replying me with these helpful information.
I double check the official document via https://docs.openwebui.com/getting-started/env-configuration
These two environment variables made 'False' as the default value.
Even, I ran the command as below to re-try:

docker run -itd -p 8080:8080 --add-host=host.docker.internal:host-gateway -e WHISPER_MODEL_AUTO_UPDATE=False -e RAG_EMBEDDING_MODEL_AUTO_UPDATE=False -v /data:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main

The container restarted always due to the error as 'Fail to connect https://huggingface.co/' (There is no internel with my product server!)

It is appreciated that you may help to review my issue, any feedback is greatful.

BR
Kimi

**Comment by quzhixue-Kimi** [comment]
Hi there,

Thank you for replying me with these helpful information.
I double check the official document via https://docs.openwebui.com/getting-started/env-configuration
These two environment variables made 'False' as the default value.
Even, I ran the command as below to re-try:

docker run -itd -p 8080:8080 --add-host=host.docker.internal:host-gateway -e  WHISPER_MODEL_AUTO_UPDATE=False -e RAG_EMBEDDING_MODEL_AUTO_UPDATE=False -v /data:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main

The container restarted always due to the error as 'Fail to connect https://huggingface.co'  (There is no internel with my product server!)
![webwxgetmsgimg](https://github.com/open-webui/open-webui/assets/8235746/813c960d-5b69-440a-a1f5-09a821f944af)

It is appreciated that you may help to review my issue, any feedback is greatful.

BR
Kimi

**Comment by cheahjs** [comment]
We pre-load the Whisper and embedding models at build time into `/app/backend/data/cache`. By using a bind mount instead of a volume with `-v /data:/app/backend/data`, this [obscures the existing cache](https://docs.docker.com/storage/bind-mounts/#mount-into-a-non-empty-directory-on-the-container), hence the need to connect to HuggingFace to fetch models.

If you need to use a bind mount, you will want to take a copy of the existing `/app/backend/data` directory with something like `docker run -it -v /data:/data ghcr.io/open-webui/open-webui:main cp -r /app/backend/data/. /data/`

**Reply by quzhixue-Kimi** [reply]
Hi there,

Thank you very much. With your suggestion, my issue has been fixed.

Thank you for your great help again.

BR
Kimi
